src/Spago/Psa.purs

```diff
@@ -144,23 +145,24 @@ toPathDecisions { rootPath, allDependencies, selectedPackages, psaCliFlags, cens
 toWorkspacePackagePathDecision
   :: { selected :: WorkspacePackage
      , psaCliFlags :: PsaOutputOptions
+     , censorProjectWarnings :: Maybe Core.CensorBuildWarnings
      }
   -> Effect (Array (LocalPath -> Maybe PathDecision))
-toWorkspacePackagePathDecision { selected: { path, package }, psaCliFlags } = do
+toWorkspacePackagePathDecision { selected: { path, package }, psaCliFlags, censorProjectWarnings } = do
   let srcPath = path </> "src"
   let testPath = path </> "test"
   pure
     [ toPathDecision
         { pathIsFromPackage: (srcPath `Path.isPrefixOf` _)
         , pathType: IsSrc
         , strict: fromMaybe false $ psaCliFlags.strict <|> (package.build >>= _.strict)
-        , censorWarnings: package.build >>= _.censorProjectWarnings
+        , censorWarnings: (package.build >>= _.censorProjectWarnings) <|> censorProjectWarnings
         }
     , toPathDecision
         { pathIsFromPackage: (testPath `Path.isPrefixOf` _)
         , pathType: IsSrc
         , strict: fromMaybe false $ psaCliFlags.strict <|> (package.test >>= _.strict)
-        , censorWarnings: package.test >>= _.censorTestWarnings
+        , censorWarnings: (package.test >>= _.censorTestWarnings) <|> censorProjectWarnings
```

Comment:
_to discuss_ I looked into pulling this out into a separate param like `censorTestWarnings`, but maybe it doesn't make sense having it under `buildOpts` as this is akin to `build:` in the package `spago.yaml`, maybe we'd need `testOpts` as `censorTestWarnings` is under `test:`.

If you get me

Reply:
I think we should have `censorTestWarnings` under `buildOpts`. I now realise that `censorTestWarnings` is misplaced under `test`, since it's a build option.

Reply:
Thanks @f-f - have moved it under `buildOpts`.
I think moving `test.censorTestWarnings` to `build.censorTestWarnings` is probably out of scope for this PR tho, sounds like a breaking change?

Reply:
> I think moving test.censorTestWarnings to build.censorTestWarnings is probably out of scope for this PR tho, sounds like a breaking change?

It doesn't need to be a breaking change if we migrate the config as we do [here](https://github.com/purescript/spago/blob/1060f4654b16047c3a74e34e5f0be1fd5d17b9ab/src/Spago/Config.purs#L652). But up to you, let's open a new ticket to track that if you would prefer to have it in a different patch

Reply:
Created a ticket to track it [here](https://github.com/purescript/spago/issues/1322).
I don't mind having a go at this once this PR is in - need to wrap my head around the config migration logic a bit first.

Reply:
Sounds great!